File: MATTHEW/leetcode/AIcoding/4.GPT-4o/c_solutions/2717-semi-ordered-permutation.c
int semiOrderedPermutation(int* nums, int numsSize) {
    int first = -1, last = -1;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 1) first = i;
        if (nums[i] == numsSize) last = i;
    }
    int result = 0;
    if (first < last) {
        result = first + (numsSize - 1 - last);
    } else {
        result = first + (numsSize - 1 - last) - 1;
    }
    return result;
}

File: leetcode/AIcoding/4.GPT-4o/rust_solutions/2717-semi-ordered-permutation.rs
impl Solution {
    pub fn semi_ordered_permutation(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut first = 0;
        let mut last = 0;

        for i in 0..n {
            if nums[i] == 1 {
                first = i;
            }
            if nums[i] == n as i32 {
                last = i;
            }
        }

        let mut result = 0;
        if first < last {
            result = (first + n - 2 - last) as i32;
        } else {
            result = (first + n - 1 - last) as i32;
        }

        result + 1
    }
}